stratosphere/pm/source/boot2/boot2_api.cpp

```diff
@@ -0,0 +1,294 @@
+                    /* Out of resource! */
+                    std::abort();
+                case ResultKernelOutOfMemory:
+                    /* Out of memory! */
+                    std::abort();
+                case ResultKernelLimitReached:
+                    /* Limit Reached! */
+                    std::abort();
+                default:
+                    /* We don't care about other issues. */
+                    break;
+            }
+
+            if (out_process_id) {
+                *out_process_id = process_id;
+            }
+        }
+
+        void LaunchList(const ncm::TitleId *launch_list, size_t num_entries) {
+            for (size_t i = 0; i < num_entries; i++) {
+                LaunchTitle(nullptr, ncm::MakeTitleLocation(launch_list[i], ncm::StorageId::NandSystem), 0);
+            }
+        }
+
+        bool GetGpioPadLow(GpioPadName pad) {
+            GpioPadSession button;
+            if (R_FAILED(gpioOpenSession(&button, pad))) {
+                return false;
+            }
+
+            /* Ensure we close even on early return. */
+            ON_SCOPE_EXIT { gpioPadClose(&button); };
+
+            /* Set direction input. */
+            gpioPadSetDirection(&button, GpioDirection_Input);
+
+            GpioValue val;
+            return R_SUCCEEDED(gpioPadGetValue(&button, &val)) && val == GpioValue_Low;
+        }
+
+        bool IsMaintenanceMode() {
+            /* Contact set:sys, retrieve boot!force_maintenance. */
+            DoWithSmSession([&]() {
+                R_ASSERT(setsysInitialize());
+            });
+            {
+                ON_SCOPE_EXIT { setsysExit(); };
+
+                u8 force_maintenance = 1;
+                setsysGetSettingsItemValue("boot", "force_maintenance", &force_maintenance, sizeof(force_maintenance));
+                if (force_maintenance != 0) {
+                    return true;
+                }
+            }
+
+            /* Contact GPIO, read plus/minus buttons. */
+            DoWithSmSession([&]() {
+                R_ASSERT(gpioInitialize());
+            });
+            {
+                ON_SCOPE_EXIT { gpioExit(); };
+
+                return GetGpioPadLow(GpioPadName_ButtonVolUp) && GetGpioPadLow(GpioPadName_ButtonVolDown);
+            }
+        }
+
+        void WaitForMitm(const char *service) {
+            const auto name = sts::sm::ServiceName::Encode(service);
+
+            while (true) {
+                bool mitm_installed = false;
+                R_ASSERT(sts::sm::mitm::HasMitm(&mitm_installed, name));
+                if (mitm_installed) {
+                    break;
+                }
+                svcSleepThread(1'000'000ull);
+            }
+        }
+
+        void LaunchFlaggedProgramsFromSdCard() {
+            /* Allow for user-customizable programs. */
+            DIR *titles_dir = opendir("sdmc:/atmosphere/titles");
+            struct dirent *ent;
+            if (titles_dir != NULL) {
```

Comment:
(Just a reminder for changing this to nullptr, unless you want to rewrite this further?)

Reply:
Thank you :)